llvm/lib/Target/AMDGPU/AMDGPULateCodeGenPrepare.cpp

```diff
@@ -546,7 +546,9 @@ class AMDGPULateCodeGenPrepareLegacy : public FunctionPass {
     AU.addRequired<TargetPassConfig>();
     AU.addRequired<AssumptionCacheTracker>();
     AU.addRequired<UniformityInfoWrapperPass>();
-    AU.setPreservesAll();
+    // This pass makes changes that can invalidate Uniformity Analysis,
+    // so don't setPreserveAll() here (see new PM version above).
+    AU.setPreservesCFG();
```

Comment:
My personal opinion is that this comment is distracting when coming to the code without the historical context, and it is also a bit beside the point. The analysis preservation here is in terms of categories: the pass preserves the CFG, which is obvious; it doesn't blanket-preserve anything else. So arguing about an individual analysis in the comment is a category error.

It's like when you take an exam and write something additional on your answer that isn't relevant to the question but happens to be wrong. I'm going to subtract points for that.

Reply:
I see. I agree that the "(see new PM version)" part is perhaps irrelevant, but let me lay out my reasoning for the rest of the comment. Though I'd also be fine with removing the comment ofc.

I added it because it immediately follows the actual 2 analyses that are used, i.e. the 

```c++
AU.addRequired<AssumptionCacheTracker>();
AU.addRequired<UniformityInfoWrapperPass>();
```

lines. The comment is basically intended to say "those are the analyses used by this pass, given that one of them is invalidated, a necessary condition is not fulfilled for `setPreservesAll()`, so `setPreservesCFG()` is the next-most suitable option". Also to make this clear for any future edits where using `setPreservesAll()` might be considered.

Reply:
(of course, the pass should most likely also blanket-invalidate analyses that it doesn't use, which might be reason enough to leave the comment out, if this is obvious for any future edits)

Reply:
Yes, that last follow-up comment is why I think the comment shouldn't be there.